background/background.js

```diff
@@ -185,20 +185,40 @@ const ContextMenu = {
   }
 };
 
-LivemarkUpdater.init().then(() => ContextMenu.init());
+// This value needs to be synchronously accessible in the
+// chrome.webRequest.onHeadersReceived listener below.
+let feedPreviewEnabled = true;
+async function feedPreview() {
+  feedPreviewEnabled = await Settings.getFeedPreviewEnabled();
+
+  Settings.addChangeListener(async (changes) => {
+    if (changes["settings.feedPreviewEnabled"]) {
+      feedPreviewEnabled = await Settings.getFeedPreviewEnabled();
+      console.log("feedPreviewEnabled changed", feedPreviewEnabled);
+    }
+  });
+}
+
+LivemarkUpdater.init().then(() => ContextMenu.init()).then(feedPreview);
```

Comment:
Can you make this slightly more elegant?
```suggestion
(async function() {
  await LivemarkUpdater.init();
  await ContextMenu.init();
  await FeedPreview.init();
})();
```